### PR TITLE
add transformParamsToQuery back to fix error

### PR DESCRIPTION
**Pull request by Boegie19**
### What does it do?

transformParamsToQuery added back for local all so you an populate it correctly

### Why is it needed?

to stop it from erroring if you have local all and populate

### How to test it?

check if it errors when local all and populate are both pressent

### Related issue(s)/PR(s)

fixes https://github.com/strapi/strapi/issues/17116

### CODE REVIEW

**Review by Marc-Roig — COMMENTED** [review]
nice catch, code looks good to me. Once you update the unit tests we can merge this.

File: packages/plugins/i18n/server/services/entity-service-decorator.js [hunk 1]
```diff
@@ -164,13 +165,14 @@ const decorator = (service) => ({
...
-    const { kind } = strapi.getModel(uid);
+    const { kind } = model;
 
     if (kind === 'singleType') {
       if (opts[LOCALE_QUERY_FILTER] === 'all') {
         // TODO Fix so this won't break lower lying find many wrappers
-        const wrappedParams = await this.wrapParams(opts, { uid, action: 'findMany' });
-        return strapi.db.query(uid).findMany(wrappedParams);
+        const wrappedParams = await this.wrapParams(opts, { uid, action: 'findMany' }) || {};
+        const query = transformParamsToQuery(model, wrappedParams);
```
Comment (Marc-Roig) [inline packages/plugins/i18n/server/services/entity-service-decorator.js:174]: transformParamsToQuery should receive a uid string no?

Comment (Boegie19) [reply packages/plugins/i18n/server/services/entity-service-decorator.js:174]: When I give it a uuid string it complains about not being able to find attributes

Comment (Marc-Roig) [reply packages/plugins/i18n/server/services/entity-service-decorator.js:174]: does it work though? can you populate fields?

Comment (Boegie19) [reply packages/plugins/i18n/server/services/entity-service-decorator.js:174]: Will test it

Comment (Boegie19) [reply packages/plugins/i18n/server/services/entity-service-decorator.js:174]: Tested it and no more error.

Comment (Marc-Roig) [reply packages/plugins/i18n/server/services/entity-service-decorator.js:174]: It doesn't error out and in some cases by luck it works, but transfomrParamsToQuery should receive a string with the model uid.

```js
const transformParamsToQuery = (uid, params) => {
  const schema = strapi.getModel(uid);
```

The resulting schema is undefined if you send the model object,  and in some cases it works, but there are some params (filters, some nested populate, ...),that tneed the schema to be computed.

File: packages/plugins/i18n/server/services/entity-service-decorator.js [hunk 2]
```diff
@@ -164,13 +165,14 @@ const decorator = (service) => ({
...
 
-    const { kind } = strapi.getModel(uid);
+    const { kind } = model;
 
     if (kind === 'singleType') {
       if (opts[LOCALE_QUERY_FILTER] === 'all') {
         // TODO Fix so this won't break lower lying find many wrappers
-        const wrappedParams = await this.wrapParams(opts, { uid, action: 'findMany' });
-        return strapi.db.query(uid).findMany(wrappedParams);
+        const wrappedParams = await this.wrapParams(opts, { uid, action: 'findMany' }) || {};
```
Comment (Marc-Roig) [inline packages/plugins/i18n/server/services/entity-service-decorator.js:173]:
```suggestion
        const wrappedParams = (await this.wrapParams(opts, { uid, action: 'findMany' })) || {};
```

File: packages/plugins/i18n/server/services/entity-service-decorator.js (hunk 1 above)
Comment (Marc-Roig) [reply packages/plugins/i18n/server/services/entity-service-decorator.js:174]: This filter doesn't work with this current implementation

image

File: packages/plugins/i18n/server/services/entity-service-decorator.js [hunk 3]
```diff
@@ -164,13 +176,15 @@ const decorator = (service) => ({
...
+    const { kind } = model;
 
     if (kind === 'singleType') {
       if (opts[LOCALE_QUERY_FILTER] === 'all') {
         // TODO Fix so this won't break lower lying find many wrappers
         const wrappedParams = await this.wrapParams(opts, { uid, action: 'findMany' });
-        return strapi.db.query(uid).findMany(wrappedParams);
+        const query = transformParamsToQuery(uid, wrappedParams);
+        const entities = await strapi.db.query(uid).findMany(query);
+        return this.wrapResult(entities, { uid, action: 'findMany' });
```
Comment (Marc-Roig) [inline packages/plugins/i18n/server/services/entity-service-decorator.js:187]: good catch adding the wrap result

**Pull request merged**